pkg/trafficpolicy/trafficpolicy.go

```diff
@@ -131,14 +132,28 @@ func MergeInboundPolicies(allowPartialHostnamesMatch bool, original []*InboundTr
 }
 
 // MergeOutboundPolicies merges two slices of *OutboundTrafficPolicies so that there is only one traffic policy for a given set of a hostnames
-func MergeOutboundPolicies(original []*OutboundTrafficPolicy, latest ...*OutboundTrafficPolicy) []*OutboundTrafficPolicy {
+// allowPartialHostnamesMatch is set to true when we intend to merge a traffic policy having host headers with other traffic policies
+func MergeOutboundPolicies(allowPartialHostnamesMatch bool, original []*OutboundTrafficPolicy, latest ...*OutboundTrafficPolicy) []*OutboundTrafficPolicy {
 	for _, l := range latest {
 		foundHostnames := false
 		for _, or := range original {
-			if reflect.DeepEqual(or.Hostnames, l.Hostnames) {
-				foundHostnames = true
-				mergedRoutes := mergeRoutesWeightedClusters(or.Routes, l.Routes)
-				or.Routes = mergedRoutes
+			if !allowPartialHostnamesMatch {
+				// For an traffic target outbound policy the hostnames list should fully intersect
```

Comment:
This comment has a typo, and is not clear to me. Could you drop an example here in addition to possibly rephrasing the comment.

Reply:
I have rephrased the comment to explain the scenario better. If I drop examples the comment will become too long. All example cases have been captured in the unit tests with appropriate names to represent the scenario, please refer the unit tests for better understanding